tests/Performance/Test/DocumentDictionaryCacheTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/**
+ * @file This file is part of the PdfParser library.
+ *
+ * @author  Konrad Abicht <k.abicht@gmail.com>
+ * @date    2020-06-01
+ *
+ * @author  Sébastien MALOT <sebastien@malot.fr>
+ * @date    2017-01-03
+ *
+ * @license LGPLv3
+ * @url     <https://github.com/smalot/pdfparser>
+ *
+ *  PdfParser is a pdf library written in PHP, extraction oriented.
+ *  Copyright (C) 2017 - Sébastien MALOT <sebastien@malot.fr>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this program.
+ *  If not, see <http://www.pdfparser.org/sites/default/LICENSE.txt>.
+ */
+
+namespace Tests\Smalot\PdfParser\Performance\Test;
+
+use Smalot\PdfParser\Parser;
+
+/**
+ * This test checks does a performance test with certain PDF files that extensively use
+ * the getFirstFont() method of Document.php. If Document.php correctly uses a dictionary
+ * to cache the objects inside the PDF file, then the parsing should be quick.
+ * If it does not, the parsing can be extensively slow or even crash.
+ */
+class DocumentDictionaryCacheTest extends AbstractPerformanceTest
+{
+    /**
+     * @var Parser
+     */
+    protected $parser;
+    protected $data;
+
+    public function init()
+    {
+        $this->parser = new \Smalot\PdfParser\Parser();
+
+        // load PDF file content
+        $this->data = file_get_contents('https://comserv.cs.ut.ee/home/files/Shoush_ComputerScience_2020.pdf?study=ATILoputoo&reference=76F6FAFD4C9E6981D9A434D32D2E7EE2AE9C49E7');
```

Comment:
I don't like dependencies from other websites, because if it goes offline, our tests break. Please copy the file to `samples` instead.

Furthermore, file must be free of charge and without any obligations. Are you `Mahmoud Kamel Shoush`?

Reply:
I am not. I'm an employee of the university this thesis document is from (I'm using the pdfparser to parse all the thesis documents). The PDF includes a license for the university to distribute this file. However, this does not extend to other third parties. So I don't think we can put this to the repo.

Reply:
In this case we need to find another solution, because I am inclined to not accept PDF via links (as described before). Is there a way to create an artificial PDF which also works here?